core-bundle/src/Image/ImageSizes.php

```diff
@@ -108,8 +108,18 @@ private function loadOptions(): void
         // The framework is required to have the TL_CROP options available
         $this->framework->initialize();
 
+        // Backwards compatibility
         $this->options = $GLOBALS['TL_CROP'] ?? [];
 
+        if (
+            3 !== \count($this->options)
+            || 0 !== \count($this->options['image_sizes'] ?? [])
+            || 2 !== \count($this->options['relative'] ?? [])
+            || 10 !== \count($this->options['exact'] ?? [])
```

Comment:
Why do we need these checks? Shouldn't `$this->options` always be an empty array if the deprecated `TL_CROP` is not used?

Reply:
`$GLOBALS['TL_CROP']` is used for reading and modifying the image size options. These checks ensure that the default value was not changed. We cannot remove the default value for `$GLOBALS['TL_CROP']` as some users will rely on it. The default is set in: https://github.com/contao/contao/blob/f2e179360dd7e1c5a10d657ca92f0746e18a0508/core-bundle/src/Resources/contao/config/config.php#L474-L491

Reply:
I see. 

So if I got you correctly, this is basically saying…
```php
$default = [
    'image_sizes' => [],
    'relative' => ['proportional', 'box'],
    'exact' => [
        'crop', 
        'left_top', 'center_top', 'right_top',
        'left_center', 'center_center', 'right_center',
        'left_bottom', 'center_bottom', 'right_bottom',
    ],
 ];

if($GLOBALS['TL_CROP'] !== $default) {
  // trigger deprecation
}`
```
…, right?

Reply:
Yes but I did not find a great way to compare them while ignoring numeric keys and the order.